src/Microsoft.TemplateEngine.Cli/TableOutput/TemplateGroupDisplay.cs

```diff
@@ -30,62 +31,90 @@ internal static IReadOnlyList<TemplateGroupTableRow> GetTemplateGroupsForListDis
+            }
+            return templateGroupsForDisplay;
+        }
 
-                languageForDisplay.Sort(StringComparer.OrdinalIgnoreCase);
-                if (!string.IsNullOrEmpty(defaultLanguageDisplay))
+        private static TemplateGroupTableRow GetTemplateGroupRow(IEnumerable<ITemplateMatchInfo> templateGroup, string language, string defaultLanguage)
+        {
+            List<string> languageForDisplay = new List<string>();
+            HashSet<string> uniqueLanguages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            string defaultLanguageDisplay = string.Empty;
+            foreach (ITemplateMatchInfo template in templateGroup)
+            {
+                string lang = template.Info.GetLanguage();
+                if (string.IsNullOrWhiteSpace(lang))
                 {
-                    languageForDisplay.Insert(0, defaultLanguageDisplay);
+                    continue;
                 }
 
-                ITemplateMatchInfo highestPrecedenceTemplate = grouping.OrderByDescending(x => x.Info.Precedence).First();
-                string shortName;
-                if (highestPrecedenceTemplate.Info is IShortNameList highestWithShortNameList)
+                if (!uniqueLanguages.Add(lang))
+                {
+                    continue;
+                }
+                if (string.IsNullOrEmpty(language) && string.Equals(defaultLanguage, lang, StringComparison.OrdinalIgnoreCase))
                 {
-                    shortName = highestWithShortNameList.ShortNameList[0];
+                    defaultLanguageDisplay = $"[{lang}]";
                 }
                 else
                 {
-                    shortName = highestPrecedenceTemplate.Info.ShortName;
+                    languageForDisplay.Add(lang);
                 }
+            }
 
-                TemplateGroupTableRow groupDisplayInfo = new TemplateGroupTableRow()
-                {
-                    Name = highestPrecedenceTemplate.Info.Name,
-                    ShortName = shortName,
-                    Languages = string.Join(",", languageForDisplay),
-                    Classifications = highestPrecedenceTemplate.Info.Classifications != null ? string.Join("/", highestPrecedenceTemplate.Info.Classifications) : null,
-                    Author = highestPrecedenceTemplate.Info.Author,
-                    Type = highestPrecedenceTemplate.Info.GetTemplateType() ?? string.Empty
-                };
-                templateGroupsForDisplay.Add(groupDisplayInfo);
+            languageForDisplay.Sort(StringComparer.OrdinalIgnoreCase);
+            if (!string.IsNullOrEmpty(defaultLanguageDisplay))
+            {
+                languageForDisplay.Insert(0, defaultLanguageDisplay);
             }
 
-            return templateGroupsForDisplay;
+            ITemplateMatchInfo highestPrecedenceTemplate = templateGroup.OrderByDescending(x => x.Info.Precedence).First();
+            string shortName;
```

Comment:
minor: nice opportunity for ternary operator use.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator